### PR TITLE
Release 2.48.0-rc0

### PR DESCRIPTION
**Pull request by LeviHarrison**
~Waiting for 10/10 Go release~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # Changelog
 
+## 2.48.0-rc.0 / 2023-10-10
+
+* [CHANGE] Remote-write: respect Retry-After header on 5xx errors. #12677
```
Comment (LeviHarrison) [inline CHANGELOG.md:5]: Wasn't sure about calling this a change or an enhancement, but it does change the current behavior of remote-write retries

**Comment by roidelapluie** [comment]
did we update the dependencies?

**Comment by LeviHarrison** [comment]
Was waiting on the go release to do that all in one

File: CHANGELOG.md (hunk 1 above)
Comment (bboreham) [reply CHANGELOG.md:5]: Could see it as a bugfix, even.

Comment (LeviHarrison) [reply CHANGELOG.md:5]: Yeah, I just don't know how expected that behavior is. Especially as its not a configuration option. But, I'm up to whatever you want.

Also on another note I see in that PR there was some discussion around documenting it, but I'm not sure if that that ever happened?

Comment (roidelapluie) [reply CHANGELOG.md:5]: it's a change to me

**Review by roidelapluie — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+* [ENHANCEMENT] Remote-write: Add a header to count retried remote write requests. #12729
+* [ENHANCEMENT] TSDB: Improve query performance by re-using iterator when moving between series. #12757
+* [ENHANCEMENT] UI: Move /targets page discovered labels to expandable section #12824
+* [ENHANCEMENT] TSDB: Optimize WBL loading by not sending empty buffers over channel. #12808
+* [ENHANCEMENT] TSDB: Reply WBL mmap markers concurrently. #12801
+* [ENHANCEMENT] Promtool: Add support for specifying series matchers in the TSDB analyze command. #12842
+* [ENHANCEMENT] PromQL: Prevent Prometheus from overallocating memory on subquery with large amount of steps. #12734
+* [ENHANCEMENT] PromQL: Add warning when monotonicity is forced in the input to histogram_quantile. #12931
+* [ENHANCEMENT] Scraping: Optimize sample appending by reducing garbage. #12939
+* [ENHANCEMENT] Storage: Reduce memory allocations when merging series sets. #12938
```
Comment (bboreham) [inline CHANGELOG.md:19]: s/when merging series sets/in queries that merge series sets/ ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+* [ENHANCEMENT] Agent: Allow ingestion of out-of-order samples. #12897
+* [ENHANCEMENT] Promtool: Improve support for native histograms in TSDB analyze command. #12869
+* [BUGFIX] SD: Ensure that discovery managers are properly canceled. #10569
+* [BUGFIX] TSDB: Fix PostingsForMatchers race with creating new series. #12558
+* [BUGFIX] TSDB: Fix handling of explicit counter reset header in histograms. #12772
+* [BUGFIX] SD: Validate HTTP client configuration in HTTP, EC2, Azure, Uyuni, PuppetDB, and Lightsail SDs. #12762 #12811 #12812 #12815 #12814 #12816
+* [BUGFIX] TSDB: Fix counter reset edgecases causing native histogram panics. #12838
+* [BUGFIX] TSDB: Fix duplicate sample detection at chunk size limit. #12874
+* [BUGFIX] Promtool: Fix errors not being reported in check rules command. #12715
+* [BUGFIX] TSDB: Register metrics after Head is initialized. #12876
```
Comment (bboreham) [inline CHANGELOG.md:31]: "Avoid panics reported in logs when head initialization takes a long time." ?

**Review by bboreham — APPROVED** [review]
I had a couple of thoughts about making the text more widely understood.
Not critical.

**Comment by roidelapluie** [comment]
Lgtm

**Review by bboreham — APPROVED** [review]

**Comment by zenador** [comment]
Note that NewPossibleNonCounterInfo in https://github.com/prometheus/prometheus/pull/12152 makes rate queries inefficient, needs fixing before release.

**Comment by juliusv** [comment]
Addendum to doing it statically: It wouldn't work in the case where you don't equality-match on the metric name, but at least you could check it once after resolving series identities for selectors, not for each resolution step?

**Comment by zenador** [comment]
Yes I agree, that particular warning and another one can be derived from the query itself, so we will move it out, but until then, we can remove it for now for a quick fix. However, other warnings do actually look at the data and so those will need to be applied every step.

**Comment by bboreham** [comment]
Some analysis of Prombench on this PR, plus meta-analysis.

I think the problem is clearest in the PromQL timings - it's 10-20% slower:
image

And once you know what's happening you can see it in memory allocations:
image

However CPU is unchanged?  What gives?
image

I believe this is because the load-generator is configured to fire queries faster than they can execute, resulting in each Prometheus doing as much work as it can, and the slower one doing less work.  
You can just about make that out in the requests/sec (note the y-axis is incorrectly labeled):
image

**Pull request merged**